apps/console/src/features/identity-providers/components/forms/components/common-pluggable-component-form.tsx

```diff
@@ -222,45 +248,61 @@ export const CommonPluggableComponentForm: FunctionComponent<CommonPluggableComp
         const bucket: ReactElement[] = [];
 
         metaProperties?.forEach((metaProperty: CommonPluggableComponentMetaPropertyInterface) => {
-
             // Ignoring elements with empty display name.
             // Note: This will remove the element from all API requests as well. If an element that is required
             // by the API is removed, that will break the app. In that case, metadata API needs to updated to
             // send a displayName for such required elements.
             if (!isEmpty(metaProperty?.displayName)) {
                 const property: CommonPluggableComponentPropertyInterface = dynamicValues?.properties?.find(
-                    property => property.key === metaProperty.key);
+                    (property: CommonPluggableComponentPropertyInterface) => property.key === metaProperty.key);
                 let field: ReactElement;
 
                 if (!isCheckboxWithSubProperties(metaProperty)) {
-                    if (metaProperty?.key === CommonConstants.FIELD_COMPONENT_SCOPES 
+
+                    if (metaProperty?.key === CommonConstants.FIELD_COMPONENT_SCOPES
                         && !isScopesDefined() && isScopesEmpty()) {
                         const updatedProperty: CommonPluggableComponentPropertyInterface = {
                             key: CommonConstants.FIELD_COMPONENT_SCOPES,
-                            value: metaProperty.defaultValue
+                            value: metaProperty.description
                         };
 
                         field = getField(updatedProperty, metaProperty, isSub,
-                            `${ testId }-form`, handleParentPropertyChange);
-                    } else {
-                        field = getField(property, metaProperty, isSub, `${ testId }-form`, handleParentPropertyChange);
-                    }                  
-                } else if (isRadioButtonWithSubProperties(metaProperty)) {
+                            `${ testId }-form`, handleParentPropertyChange, manualMode);
+                    }
+                    else if (
+                        metaProperty?.key === "SelectMode"
+                        || metaProperty?.key ==="SPEntityId"
+                        || metaProperty?.key === "NameIDType"
```

Comment:
Instead of chaining OR conditions, we might be able to do something like this for better readability.

```suggestion
                        ["SelectMode", "SPEntityId", "NameIDType"].includes(metaProperty?.key)
```

Reply:
Improved with 769cf8391b5176fdb1cbee9cf4cc3fe82c8057d4.